Lua/Lua.sublime-syntax

```diff
@@ -75,10 +85,34 @@ contexts:
     - include: line-comments
 
   block-comments:
+    - match: --\[\[--
+      scope: punctuation.definition.comment.begin.lua
+      push: block-doc-comment-body
     - match: --\[(=*)\[
       scope: punctuation.definition.comment.begin.lua
       push: block-comment-body
 
+  block-doc-comment-body:
+    - meta_include_prototype: false
+    - meta_scope: comment.block.documentation.lua
+    - match: \]\]
+      scope: punctuation.definition.comment.end.lua
+      pop: 1
+    - match: ^\s*(@section)\s+(.*?\S)\s*(?=\]\]|$)
+      captures:
+        1: entity.name.tag.documentation.lua
+        2: entity.name.section.lua
+    - match: ^\s*(@param)\s+({{identifier}})\b
```

Comment:
Maybe add a type highlight?

Reply:
The difficulty is that the tags are not really standardized. There are also [LDoc](https://stevedonovan.github.io/ldoc/manual/doc.md.html#Commenting_Conventions) and LuaDoc, which use a slightly different convention. If we add type highlighting for the next word after the parameter name, then for LDoc it would mistakenly highlight the first word of the description as a type. By keeping it simple, we can reduce the chance of false highlighting for all users. Would you say that EmmyLua is the de-facto standard?

From LDoc:

![LDoc](https://user-images.githubusercontent.com/6579999/153742268-217d0c10-388c-4775-9487-29a5ecf90be8.png)

Reply:
Emmmm,not thought about that :cold_sweat:

Reply:
So far everything is fine :smiley: